From ca2ba5aeb3eef8e6e59de2bac42d76b25a47689b Mon Sep 17 00:00:00 2001
From: Manoj Sai <abbaraju.manojsai@amarulasolutions.com>
Date: Thu, 7 Jul 2022 14:53:12 +0530
Subject: [PATCH 4/6] os_dep/recv_linux: removed GRO_DROP macro check condition

GRO_DROP macro support is removed in latest version kernel
https://github.com/cilynx/rtl88x2bu/issues/157

Signed-off-by: Manoj Sai <abbaraju.manojsai@amarulasolutions.com>
Change-Id: Ia8a02e38dfd3be30048d1385146eb42327c787a9
---
 os_dep/linux/recv_linux.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/os_dep/linux/recv_linux.c b/os_dep/linux/recv_linux.c
index 2b355f2..aeec9b5 100644
--- a/os_dep/linux/recv_linux.c
+++ b/os_dep/linux/recv_linux.c
@@ -355,8 +355,13 @@ static int napi_recv(_adapter *padapter, int budget)
 
 #ifdef CONFIG_RTW_GRO
 		if (pregistrypriv->en_gro) {
+			#if (LINUX_VERSION_CODE >= KERNEL_VERSION(5, 12, 0))
+			rtw_napi_gro_receive(&padapter->napi, pskb);
+			rx_ok = _TRUE;
+			#else
 			if (rtw_napi_gro_receive(&padapter->napi, pskb) != GRO_DROP)
 				rx_ok = _TRUE;
+			#endif
 			goto next;
 		}
 #endif /* CONFIG_RTW_GRO */
-- 
2.25.1

